core/services/vrf/listener_v2.go

```diff
@@ -295,27 +295,37 @@ func (lsn *listenerV2) processRequestsPerSub(
 		return
 	}
 	lggr := lsn.l.With(
-		"sub", reqs[0].req.SubId,
+		"subID", reqs[0].req.SubId,
 		"maxGasPrice", maxGasPrice.String(),
 		"reqs", len(reqs),
 		"startBalance", startBalance.String(),
 		"startBalanceNoReservedLink", startBalanceNoReserveLink.String(),
 	)
 	lggr.Infow("Processing requests for subscription")
+
 	// Attempt to process every request, break if we run out of balance
 	var processed = make(map[string]struct{})
+	// Keep track of the number of times we process each request - this could be > 1
+	// especially if we run into re-orgs. This is exclusively for logging purposes.
+	uniqueRequests := map[string]int{}
 	for _, req := range reqs {
+		uniqueRequests[req.req.RequestId.String()]++
```

Comment:
this loop is already pretty overloaded with logic - maybe just pull out a little helper that converts reqs into a set and then just call that in the final "finished processing" log?